fvm/transactionVerifier.go

```diff
@@ -184,6 +185,14 @@ func (v *TransactionSignatureVerifier) verifyAccountSignature(
 		return nil, errors.NewInvalidPayloadSignatureError(txSig.Address, txSig.KeyIndex, err)
 	}
 
+	if !v.accountSignatureSchemeIsSupported(accountKey) {
```

Comment:
So I looked further at the code and I noticed that my previous comment wasn't totally valid: when we are in the transaction signature case, we fetch the account public key from the state, that key is guaranteed to have the correct hashing algorithm, otherwise it wouldn't have been added to the state. You can see the check done [here](https://github.com/onflow/flow-go/blob/master/fvm/state/accounts.go#L274-L282). 
This means that we do not need to add this extra check and the original code is fine.

Reply:
great, going to revert it then